woocommerce/woocommerce-ios
# Add warning UI to variation list cell details and variation form

Most parts for #2594 

This PR added warning UI for the following parts:

### Product variation form

- If a variation is enabled but doesn't have a price set, let's update the following:
  - update the price row to be warning color "Add Price" 
  - add a warning banner row below

<img width="504" alt="Screen Shot 2020-08-06 at 10 18 54 AM" src="https://user-images.githubusercontent.com/1945542/89483317-5951f680-d7ce-11ea-98aa-c470a7fbf1cc.png">

### Variations list

(The top banner will be implemented in a separate PR)
- Update variation cell details text based on the follow states:
  - Variation is disabled: "Disabled" is shown
  - Variation is enabled and has a price: the price is shown as before
  - Variation is enabled but doesn't have a price: "No price set" is shown in warning color

<img width="390" alt="Screen Shot 2020-08-07 at 1 35 43 PM" src="https://user-images.githubusercontent.com/1945542/89612663-ef5f4d00-d8b2-11ea-9fb4-87db924dca5e.png">


## Changes

### Changes for product variation form
- Created a new icon `UIImage.infoOutlineImage` with the outline info Gridicons type
- Semantic color updates:
  - Updated the existing `warning` color to be the tint color of the warning text (orange 30/50 in light/dark mode)
  - Added a new semantic color `warningBackground` color to be a lighter version of the `warning` color
- Added two helpers to `EditableProductVariationModel`:
  - `isEnabled: Bool`: used for the status switch and for the following helper
  - `isEnabledAndMissingPrice: Bool`: used for showing the warning UI on the no price row and the price row tint color
- Added `noPriceWarning` to variation form (`ProductFormEditAction` and `ProductFormSection.SettingsRow`) which is shown when a variation is enabled but has no price. The new row configures `ImageAndTitleAndTextTableViewCell` with a new view model `WarningViewModel` that takes in an icon and title text.
- Updated `ProductVariationFormActionsFactory` to show `noPriceWarning` row when a variation's `isEnabledAndMissingPrice` is true

### Changes for variations list - cell details text

- Updated `ProductsTabProductViewModel.init(productVariationModel:...)` so that the details text now shows different value based on the following 3 states:
  - Variation is disabled: "Disabled" is shown
  - Variation is enabled and has a price: the price is shown as before
  - Variation is enabled but doesn't have a price: "No price set" is shown in warning color
- In `ImageAndTitleAndTextTableViewCell`, added `tintColor: UIColor?` so that when it's non-nil, the tint color is applied to icon and labels for the no price state
- Also DI'ed `CurrencySettings` to `ProductsTabProductViewModel.init(productVariationModel:...)` so that the case where a variation is enabled & price set can be tested

## Testing

- Go to the Products tab
- Tap on a variable product
- Tap on the variations row --> the cell's details text should follow the design spec as in #2594 (3 states: Disabled, Price, No price) - feel free to check this screen after updating a variation in later steps
- Tap on a variation
- Tap on the price row to remove the price, if it currently has a price, and go back to the variation screen
- Turn on the "Enabled" switch --> a warning row should appear as in design and the price row is in warning color
- Turn off the "Enabled" switch --> the warning row should disappear
- Turn on the "Enabled" switch again  --> the warning row should appear again and the price row is in warning color
- Tap on the price row to set a price --> the warning row should disappear

## Example screenshots

Note: the top banner on the variation list will be implemented separately

I do love how clear the status for each variation on the variation list is now!

![simulator](https://user-images.githubusercontent.com/1945542/89616362-d9ee2100-d8ba-11ea-9f59-2ff9b18cd01d.gif)

\ | light | dark
-- | -- | --
Variation list | ![Simulator Screen Shot - iPhone 11 - 2020-08-07 at 13 41 13](https://user-images.githubusercontent.com/1945542/89616344-d22e7c80-d8ba-11ea-8de0-6083791ff7dc.png) | ![Simulator Screen Shot - iPhone 11 - 2020-08-07 at 13 41 23](https://user-images.githubusercontent.com/1945542/89616354-d65a9a00-d8ba-11ea-8236-349cbca8c4f0.png)
Variation form (variation is enabled but has no price) | ![Simulator Screen Shot - iPhone 11 - 2020-08-07 at 13 41 36](https://user-images.githubusercontent.com/1945542/89616356-d78bc700-d8ba-11ea-942a-0dc42d0c6173.png) | ![Simulator Screen Shot - iPhone 11 - 2020-08-07 at 13 41 48](https://user-images.githubusercontent.com/1945542/89616358-d8bcf400-d8ba-11ea-8764-c42ec27b7e7c.png)





Update release notes:
